roles/rpm_ostree_status/tasks/main.yml

```diff
@@ -11,16 +11,27 @@
 
 - name: Convert to JSON
   set_fact:
-    ros_json: "{{ ros.stdout|from_json }}"
+    ros_json: "{{ ros.stdout | from_json }}"
 
-- name: Set ros variable if deployment 0 is booted
+- name: Set ros_num_deployments
   set_fact:
-    ros_booted: "{{ ros_json['deployments'][0] }}"
-    ros_not_booted: "{{ ros_json['deployments'][1] if ros_json['deployments'][1] is defined else false }}"
-  when: ros_json['deployments'][0] is defined and ros_json['deployments'][0]['booted']
+    ros_num_deployments: "{{ ros_json|json_query('length(deployments)') }}"
+
+- block:
+  - name: Set ros_booted variable for multi-deployments
+    set_fact:
+      ros_booted: "{{ item }}"
+    with_items: "{{ ros_json | json_query('deployments[?booted]') }}"
+
+  - name: Set ros_not_booted variable
+    set_fact:
+      ros_not_booted: "{{ item | default(false) }}"
+    with_items: "{{ ros_json | json_query('deployments[?!booted]') }}"
+  when: ros_num_deployments > 1
```

Comment:
I'm surprised this block doesn't work equally well when `ros_num_booted == 1`. Is that not the case?

Reply:
Yeah, your logic is better than mine!  😁

Reply:
Oh, I remember why I did it like this.  When the user does `rpm-ostree ex livefs`, we end up with three deployments and I was trying to determine the 'not-booted' one.  But we actually have two 'not-booted', so this is wrong anyways.  😬